### PR TITLE
TLF-106 COA - Notify email to the business/caseworker and user

### PR DESCRIPTION
**Pull request by MikeMaratheftis**
## What? 
TLF-106 - COA - Notify email to the business/caseworker and user [TLF-106](https://collaboration.homeoffice.gov.uk/jira/browse/TLF-106)
## Why? 
Allow user and business to receive a confirmation email on form submition.
## How? 
Confirmation emails are sent using the NotifyClient from the [notifications-node-client](https://www.npmjs.com/package/notifications-node-client) package. This is done in the sendUserEmailNotification and sendCaseworkerEmailNotification methods of the SendEmailConfirmation class.
## Check list

- [x] I have reviewed my own pull request
- [ ] I have written tests (if relevant)

### CODE REVIEW

File: apps/coa/behaviours/send-email-notification.js [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+const notifyKey = config.govukNotify.notifyApiKey;
+const translation = require('../translations/src/en/fields.json');
+const notifyClient = new NotifyClient(notifyKey);
+
+module.exports = class SendEmailConfirmation {
+  
+  async sendUserEmailNotification(req) {
+    const personalisations = this.notifyPersonalisations;
+
+    // Retrieve the value from the session model
```
Comment (dk4g) [inline apps/coa/behaviours/send-email-notification.js:18]: No need for this comment, the code is self-explanatory enough.

File: apps/coa/behaviours/send-email-notification.js [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+          new_home_address: req.sessionModel.get('addressDetails') ?? '',
+          has_new_postal_address: req.sessionModel.get('steps').includes('/postal-address') ? 'yes' : 'no',
+          new_postal_address: req.sessionModel.get('postalAddressDetails') ?? '',
+        })
+      });
+
+      const trackedPageStartTime = Number(req.sessionModel.get('session.started.timestamp'));
+      const timeSpentOnForm = utilities.secondsBetween(trackedPageStartTime, new Date());
+
+      req.log('info', 'coa.submit_form.create_email_with_file_notify.successful');
```
Comment (dk4g) [inline apps/coa/behaviours/send-email-notification.js:124]: The log message is not clear enough, I would recommend modifying it for readability and to include additional information about the type of email sent - `User confirmation` or `Business confirmation`, and perhaps  a reference number?)

Something like `User confirmation has been successfully sent. Reference number: <>`

File: apps/coa/behaviours/send-email-notification.js [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+module.exports = class SendEmailConfirmation {
+  
+  async sendUserEmailNotification(req) {
+    const personalisations = this.notifyPersonalisations;
+
+    // Retrieve the value from the session model
+    const details_updating_value = req.sessionModel.get('which-details-updating');
+    
+    // Retrieve the corresponding label from the translations file
+    const details_updating_label = translation['which-details-updating'].options[details_updating_value].label;
```
Comment (dk4g) [inline apps/coa/behaviours/send-email-notification.js:22]: `which-details-updating` can return string or array if multiple choices were made

File: apps/coa/behaviours/send-email-notification.js [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+
+    try {
+      if (notifyKey === 'USE_MOCK') {
+        req.log('warn', '*** Notify API Key set to USE_MOCK. Ensure disabled in production! ***');
+      }
+
+      await notifyClient.sendEmail(userConfirmationTemplateId, req.sessionModel.get('email'), {
+        personalisation: Object.assign({}, personalisations, {
+        reference_number: req.sessionModel.get('uniqueRefNumber'),
+        applying_for_whom: req.sessionModel.get('isApplicant') ? 'your' : req.sessionModel.get('nameWithPossession'),
```
Comment (vinodhasamiyappanHO) [inline apps/coa/behaviours/send-email-notification.js:32]: `applying_for_whom` - is not used in user template. replace the `applicant_full_name `with `applying_for_whom` placeholder

You requested to update ((applicant_full_name)) details
Thank you for your request to change ((applicant_full_name)) details with UK Visas and Immigration.

File: apps/coa/behaviours/send-email-notification.js [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+        brp_number: req.sessionModel.get('brp-details'),
+        has_arc_number: req.sessionModel.get('arc-details') ? 'yes' : 'no',
+        arc_number: req.sessionModel.get('arc-details'),
+        who_are_you: req.sessionModel.get('who-are-you'),
+        is_legal_rep: req.sessionModel.get('legal-representative-name') ? 'yes' : 'no',
+        leg_rep_full_name: req.sessionModel.get('legal-representative-name'),
+        leg_rep_email_address: req.sessionModel.get('email'),
+        has_rep_tel_number: req.sessionModel.get('telephone').length > 0 ? 'yes' : 'no',
+        leg_rep_tel_number: req.sessionModel.get('telephone'),
+        has_leg_details: req.sessionModel.get('steps').includes('/legal-details') && req.sessionModel.get('oisc-sra-number') ? 'yes' : 'no',
```
Comment (vinodhasamiyappanHO) [inline apps/coa/behaviours/send-email-notification.js:50]: This condition can only check the steps for `/legal-details. If the steps not included the field value will not be there. so `oisc-sra-number` can be removed.
req.sessionModel.get('steps').includes('/legal-details') ? 'yes' : 'no',

File: apps/coa/behaviours/send-email-notification.js [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+          brp_number: req.sessionModel.get('brp-details'),
+          has_arc_number: req.sessionModel.get('arc-details') ? 'yes' : 'no',
+          arc_number: req.sessionModel.get('arc-details'),
+          is_legal_rep: req.sessionModel.get('legal-representative-name') ? 'yes' : 'no',
+          leg_rep_full_name: req.sessionModel.get('legal-representative-name'),
+          who_are_you: req.sessionModel.get('who-are-you'),
+          leg_rep_address: req.sessionModel.get('legalAddressDetails') ?? '',
+          has_rep_tel_number: req.sessionModel.get('telephone').length > 0 ? 'yes' : 'no',
+          leg_rep_tel_number: req.sessionModel.get('telephone'),
+          has_leg_details: req.sessionModel.get('steps').includes('/legal-details') && req.sessionModel.get('oisc-sra-number') ? 'yes' : 'no',
```
Comment (vinodhasamiyappanHO) [inline apps/coa/behaviours/send-email-notification.js:107]: This condition can only check the steps for `/legal-details`. If the steps not included the field value will not be there. so `oisc-sra-number` can be removed.
req.sessionModel.get('steps').includes('/legal-details') ? 'yes' : 'no',

**Review by vinodhasamiyappanHO — APPROVED** [review]
Looks good to me

**Pull request merged**